### PR TITLE
Wrap n_factor and implement factor(::Int) and factor(::UInt)

### LINKED ISSUE #970: Wrap `n_factor`
And use it in `divisors`.

### PR DESCRIPTION
**Pull request by wbhart**
Fixes #970

### CODE REVIEW

File: src/flint/fmpz.jl [hunk 1]
```diff
@@ -1165,6 +1165,22 @@ function _factor(a::fmpz)
...
 
+function factor(a::Union{Int, UInt})
+   if iszero(a)
+      throw(ArgumentError("Argument is not non-zero"))
+   end
+   u = sign(a)
+   a = u < 0 ? -a : a
+   F = n_factor()
+   ccall((:n_factor, libflint), Nothing, (Ref{n_factor}, UInt), F, a)
+   res = Dict{Int, Int}()
```
Comment (tthsqe12) [inline src/flint/fmpz.jl:1176]: To be pedantic, shouldn't the keys have the same type as `a`? So, maybe `res = Dict{typeof(a), Int}()` here?

**Review by tthsqe12 — CHANGES_REQUESTED** [review]

File: src/flint/fmpz.jl (hunk 1 above)
Comment (tthsqe12) [reply src/flint/fmpz.jl:1176]: This is more than pedantry, the current version doesn't work across the UInt range:
```
julia> factor(next_prime(3*UInt(2)^62))
ERROR: InexactError: check_top_bit(Int64, 13835058055282163729)
```

Comment (wbhart) [reply src/flint/fmpz.jl]: Should be fixed now.

File: src/flint/fmpz.jl [hunk 2]
```diff
@@ -1373,6 +1391,7 @@ function next_prime(x::fmpz, proved::Bool = true)
 end
 
 function next_prime(x::UInt, proved::Bool = true)
+   x >= 0xffffffffffffffc5 && error("No larger single-limb prime exists")
```
Comment (tthsqe12) [inline src/flint/fmpz.jl:1394]:
```suggestion
```
No. This is 64-bit specific, and it is already done correctly. See test code.

Comment (wbhart) [reply src/flint/fmpz.jl:1394]: True it is 64 bit specific, but the flint exception prints an error in our tests which shows up in the test output.

Comment (tthsqe12) [reply src/flint/fmpz.jl:1394]: This is what we want, right?

Comment (wbhart) [reply src/flint/fmpz.jl]: Hopefully correct this time. Can't check as Nemo is using the latest AA which is broken without the corresponding Nemo PR's merged.

Comment (wbhart) [reply src/flint/fmpz.jl]: Why would we want an error message to print while our tests are running?

Comment (tthsqe12) [reply src/flint/fmpz.jl]: If you don't want to see the error messages from flint printed to stdout, then the exception system in flint needs an overhaul.

Comment (wbhart) [reply src/flint/fmpz.jl]: Yes it does.

Comment (tthsqe12) [reply src/flint/fmpz.jl]: You should see the singular.jl test code: we print all of the error messages and it is fine.

**Pull request merged**